### PR TITLE
New module dynamodb_table_info

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

New info module for dynamodb table

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

dynamodb_table_info

### CODE REVIEW

File: plugins/modules/dynamodb_table_info.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: dynamodb_table_info
+version_added: 7.1.0
```
Comment (GomathiselviS) [inline plugins/modules/dynamodb_table_info.py:10]:
```suggestion
version_added: 7.2.0
```

File: plugins/modules/dynamodb_table_info.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+short_description: Returns information about a Dynamo DB table
+description:
+  - Returns information about the Dynamo DB table, including the current status of the table
```
Comment (GomathiselviS) [inline plugins/modules/dynamodb_table_info.py:13]:
```suggestion
  - Returns information about the Dynamo DB table, including the current status of the table,
```

**Review by GomathiselviS — APPROVED** [review]

File: tests/integration/targets/dynamodb_table/tasks/main.yml [hunk 3]
```diff
@@ -65,6 +75,17 @@
         - create_table.table_name == table_name
         - create_table.write_capacity == 1
 
+  - name: Ensure the table was not created
```
Comment (hakbailey) [inline tests/integration/targets/dynamodb_table/tasks/main.yml:78]:
```suggestion
  - name: Ensure the table was created
```

File: tests/integration/targets/dynamodb_table/tasks/test_pay_per_request.yml [hunk 4]
```diff
@@ -32,6 +42,17 @@
       - create_table is changed
       - create_table.billing_mode == "PAY_PER_REQUEST"
 
+- name: Ensure the table was created
+  dynamodb_table_info:
+    name: "{{ table_name_on_demand }}"
+  register: _table
+
+- name: Assert the table has been created with the expected billing modw
```
Comment (hakbailey) [inline tests/integration/targets/dynamodb_table/tasks/test_pay_per_request.yml:50]:
```suggestion
- name: Assert the table has been created with the expected billing mode
```

File: tests/integration/targets/dynamodb_table/tasks/test_pay_per_request.yml [hunk 5]
```diff
@@ -120,6 +151,17 @@
       - create_complex_table.table_name == table_name_on_demand_complex
       - create_complex_table.tags == tags_default
 
+- name: Ensure the table was not created
```
Comment (hakbailey) [inline tests/integration/targets/dynamodb_table/tasks/test_pay_per_request.yml:154]:
```suggestion
- name: Ensure the table was created
```

File: plugins/modules/dynamodb_table_info.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            sample: true
+        provisioned_throughput:
+            description: The provisioned throughput settings for the table.
+            type: dict
+            returned: always
+            sample: '{"number_of_decreases_today": 0, "read_capacity_units": 1, "write_capacity_units": 1}'
+        tags:
+            description: A dict of tags associated with the DynamoDB table.
+            returned: always
+            type: dict
```
Comment (hakbailey) [inline plugins/modules/dynamodb_table_info.py:176]: It seems like there are some values boto3 could return that aren't listed here, like `SSEDescription` and `ArchivalSummary`.

File: plugins/modules/dynamodb_table_info.py [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    except (
+        botocore.exceptions.ClientError,
+        botocore.exceptions.BotoCoreError,
+    ) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to list table tags")
+
+    table = camel_dict_to_snake_dict(table)
+    table["tags"] = boto3_tag_list_to_ansible_dict(tags)
+
+    if "table_class_summary" in table:
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/modules/dynamodb_table_info.py:230]: @hakbailey  This is done to have the same output as the one from module `dynamodb_table`

File: plugins/modules/dynamodb_table_info.py (hunk 6 above)
Comment (abikouo) [reply plugins/modules/dynamodb_table_info.py:199]: @hakbailey I have added the missing elements into the output

**Pull request merged**